pkg/traceability/sampling/globalsampling.go

```diff
@@ -40,6 +42,12 @@ func SetupSampling(cfg Sampling) error {
 
 // ShouldSampleTransaction - receives the transaction details and returns true to sample it false to not
 func ShouldSampleTransaction(details TransactionDetails) (bool, error) {
+	hasFailedStatus := details.Status == "Failure"
+	// sample the transaction if reportAllErrors is set to `true` and the trasaction summary's status is an error
+	if hasFailedStatus && agentSamples.config.ReportAllErrors {
+		return true, nil
+	}
+
```

Comment:
This new block should be in the agentSample.ShouldSampleTransaction method, not the global one.